src/pinecone.ts

```diff
@@ -201,18 +201,16 @@ export class Pinecone {
    * @param indexName - The name of the index to describe.
    * @returns A promise that resolves to {@link IndexModel}
    */
-  describeIndex(indexName: IndexName) {
-    const describeIndexPromise = this._describeIndex(indexName);
+  async describeIndex(indexName: IndexName) {
```

Comment:
This is the change to support interacting with results from the `describeIndex` call in order to push things into `IndexHostSingleton`.

Reply:
Another place you can do something similar is in `listIndexes()` since that response now includes the entire index model with host info and not just the index names.

Reply:
Good call, I've added an Asana task to pursue that as a follow up.  https://app.asana.com/0/1205585657100015/1206081872049522/f